runner.yml

```diff
@@ -338,12 +345,12 @@ Resources:
                   [runners.machine]
                     IdleCount = ${GitlabIdleCount}
                     IdleTime = ${GitlabIdleTime}
-                    MaxBuilds = ${GitlabMaxConcurrentBuilds}
+                    MaxBuilds = 100
                     MachineDriver = "amazonec2"
                     MachineName = "gitlab-docker-machine-%s"
-                    MachineOptions = ["amazonec2-instance-type=${GitlabRunnerInstanceType}", "amazonec2-region=${AWS::Region}", "amazonec2-vpc-id=${VpcId}", "amazonec2-security-group=${AWS::StackName}-RunnersSecurityGroup", "amazonec2-use-private-address=true", "amazonec2-iam-instance-profile=${RunnersInstanceProfile}"]
+                    MachineOptions = ["engine-storage-driver=overlay2", "amazonec2-instance-type=${GitlabRunnerInstanceType}", "amazonec2-region=${AWS::Region}", "amazonec2-vpc-id=${VpcId}", "amazonec2-security-group=${AWS::StackName}-RunnersSecurityGroup", "amazonec2-use-private-address=true", "amazonec2-iam-instance-profile=${RunnersInstanceProfile}", "amazonec2-request-spot-instance=true", "amazonec2-spot-price=0.08"]
```

Comment:
* `"engine-storage-driver=overlay2"` this is surely an improvement. I wonder if we should make this configurable, but it's probably good so.
* `"amazonec2-request-spot-instance=true", "amazonec2-spot-price=0.08"`: I don't like the fact that spot instances are used in any case. Somebody may still want to use on-demand instances. Also, this hardcoded bid price is good for `m5.large` instances, but it's definitely too much for, say, `t3.medium` instances.

I'd prefer to do something like this:
1. add a parameter `SpotInstancePrice` that can be any value ≥ 0 (`[0, +∞)`, or `[0, +∞[` in French/Italian notation).
2. add a condition `RequestSpotInstance` that checks whether `SpotInstancePrice > 0` — so we are requesting spot instances if and only if `SpotInstancePrice ≠ 0`.
3. Use the parameter and the condition in this template to set `amazonec2-request-spot-instance` to true or false depending on the condition's value, and `amazonec2-spot-price` to the parameter's value.

Reply:
overlay2 is the default everywhere and recommanded by gitlab, yet docker machine does make it default yet ... I guess there is only good there.

for spot I've converted it to 2 parameter a Yes/No and a price one. at least there is no explaination on the >0 price enable spot etc ...